daemons/fenced/pacemaker-fenced.c

```diff
@@ -578,7 +579,44 @@ our_node_allowed_for(pe_resource_t *rsc)
 }
 
 static void
-watchdog_device_update(xmlNode *cib)
+watchdog_device_update(void)
+{
+    if (stonith_watchdog_timeout_ms > 0) {
+        if (!g_hash_table_lookup(device_list, STONITH_WATCHDOG_ID) &&
+            !stonith_watchdog_targets) {
+            /* getting here watchdog-fencing enabled, no device there yet
+               and reason isn't stonith_watchdog_targets preventing that
+             */
+            int rc;
+            xmlNode *xml;
+
+            xml = create_device_registration_xml(
+                    STONITH_WATCHDOG_ID,
+                    st_namespace_internal,
+                    STONITH_WATCHDOG_AGENT,
+                    NULL, /* stonith_device_register will add our
+                             own name as PCMK_STONITH_HOST_LIST param
+                             so we can skip that here
+                           */
+                    NULL);
+            rc = stonith_device_register(xml, NULL, FALSE);
+            free_xml(xml);
+            if (rc != pcmk_ok) {
+                crm_crit("Cannot register watchdog pseudo fence agent");
+                crm_exit(CRM_EX_FATAL);
+            }
+        }
+
+    } else {
+        /* be silent if no device - todo parameter to stonith_device_remove */
+        if (g_hash_table_lookup(device_list, STONITH_WATCHDOG_ID)) {
+            stonith_device_remove(STONITH_WATCHDOG_ID, FALSE);
```

Comment:
Are you referring to the silencing I had put in when it came from the CIB? Didn't I remove that again?
Otherwise I guess this deletion can happen multiple times just to assure the device is removed if stonith-watchdog-fencing is disabled. So it happens even if there is no device anymore. And even when just disabled there might not have been one before due to the host-list.

Reply:
I mean shouldn't the second argument to stonith_device_remove() be TRUE, so we clear cib-registered rather than api-registered?

Reply:
now I get it ;-)
more taking over what was there before. yeah I guess TRUE would be better as it is just called via the cib-init/update.
changed it